addon/components/variable-plugin/codelist/edit.ts

```diff
@@ -81,11 +80,12 @@ export default class CodelistEditComponent extends Component<Args> {
   }
 
   codelistOptions = trackedFunction(this, async () => {
+    let result: CodeListOptions | undefined;
     if (this.source && this.codelistUri) {
-      return fetchCodeListOptions(this.source, this.codelistUri);
-    } else {
-      return;
+      result = await fetchCodeListOptions(this.source, this.codelistUri);
     }
+    this.selectedCodelistOption = undefined;
```

Comment:
I'd like a comment here to explain this (= resources with side-effects) is really not a pattern we should internalize, and it's basically a hack that nullvox gave us

Reply:
otherwise good to go

Reply:
Good point.